From 41b65ef595e7728ab8ba7d96b50ee22753f77d6d Mon Sep 17 00:00:00 2001
From: Meir Levi <mlevi4@marvell.com>
Date: Wed, 4 Nov 2020 04:51:26 -0500
Subject: [PATCH 1071/1921] dts: marvell: musdk: add generic UIO node for pp2
 registers

Patch provide access to MUSDK pp2 registers space.

Signed-off-by: Meir Levi <mlevi4@marvell.com>
Change-Id: Ie1b275ff5b153fc910dd7804f93c8cd844486a70
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/42556
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Tested-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-70x0.dtsi  | 2 ++
 arch/arm64/boot/dts/marvell/armada-80x0.dtsi  | 4 ++++
 arch/arm64/boot/dts/marvell/armada-cp11x.dtsi | 7 +++++++
 arch/arm64/boot/dts/marvell/cn9130.dtsi       | 2 ++
 arch/arm64/boot/dts/marvell/cn9131-db.dts     | 2 ++
 arch/arm64/boot/dts/marvell/cn9132-db.dts     | 2 ++
 6 files changed, 19 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-70x0.dtsi b/arch/arm64/boot/dts/marvell/armada-70x0.dtsi
index 293403a1a333..839bf984e93d 100644
--- a/arch/arm64/boot/dts/marvell/armada-70x0.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-70x0.dtsi
@@ -18,6 +18,7 @@
  * Instantiate the CP110
  */
 #define CP11X_NAME		cp0
+#define CP11X_NUM		0
 #define CP11X_BASE		f2000000
 #define CP11X_PCIEx_MEM_BASE(iface) (0xf6000000 + (iface * 0x1000000))
 #define CP11X_PCIEx_MEM_SIZE(iface) 0xf00000
@@ -28,6 +29,7 @@
 #include "armada-cp110.dtsi"
 
 #undef CP11X_NAME
+#undef CP11X_NUM
 #undef CP11X_BASE
 #undef CP11X_PCIEx_MEM_BASE
 #undef CP11X_PCIEx_MEM_SIZE
diff --git a/arch/arm64/boot/dts/marvell/armada-80x0.dtsi b/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
index ee67c70bf02e..4e7729df2944 100644
--- a/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
@@ -20,6 +20,7 @@
  * Instantiate the master CP110
  */
 #define CP11X_NAME		cp0
+#define CP11X_NUM		0
 #define CP11X_BASE		f2000000
 #define CP11X_PCIEx_MEM_BASE(iface) (0xf6000000 + (iface * 0x1000000))
 #define CP11X_PCIEx_MEM_SIZE(iface) 0xf00000
@@ -30,6 +31,7 @@
 #include "armada-cp110.dtsi"
 
 #undef CP11X_NAME
+#undef CP11X_NUM
 #undef CP11X_BASE
 #undef CP11X_PCIEx_MEM_BASE
 #undef CP11X_PCIEx_MEM_SIZE
@@ -41,6 +43,7 @@
  * Instantiate the slave CP110
  */
 #define CP11X_NAME		cp1
+#define CP11X_NUM		1
 #define CP11X_BASE		f4000000
 #define CP11X_PCIEx_MEM_BASE(iface) (0xfa000000 + (iface * 0x1000000))
 #define CP11X_PCIEx_MEM_SIZE(iface) 0xf00000
@@ -51,6 +54,7 @@
 #include "armada-cp110.dtsi"
 
 #undef CP11X_NAME
+#undef CP11X_NUM
 #undef CP11X_BASE
 #undef CP11X_PCIEx_MEM_BASE
 #undef CP11X_PCIEx_MEM_SIZE
diff --git a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
index aebe5820310e..deb549f419b8 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
@@ -127,6 +127,13 @@
 			};
 		};
 
+		CP11X_LABEL(uio_ethernet): EVALUATOR(uio_pp_, CP11X_NUM)@0 {
+			compatible = "generic-uio";
+			reg = <0x0 0x90000>, <0x130000 0x6000>,
+			      <0x220000 0x1000>;
+			reg-names = "pp", "mspg", "cm3";
+		};
+
 		CP11X_LABEL(comphy): phy@120000 {
 			compatible = "marvell,comphy-cp110";
 			reg = <0x120000 0x6000>;
diff --git a/arch/arm64/boot/dts/marvell/cn9130.dtsi b/arch/arm64/boot/dts/marvell/cn9130.dtsi
index a2b7e5ec979d..0cd32d68a0e7 100644
--- a/arch/arm64/boot/dts/marvell/cn9130.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130.dtsi
@@ -18,6 +18,7 @@
  */
 
 #define CP11X_NAME		cp0
+#define CP11X_NUM		0
 #define CP11X_BASE		f2000000
 #define CP11X_PCIEx_MEM_BASE(iface) ((iface == 0) ? 0xc0000000 : \
 						    0xe0000000 + ((iface - 1) * 0x1000000))
@@ -29,6 +30,7 @@
 #include "armada-cp115.dtsi"
 
 #undef CP11X_NAME
+#undef CP11X_NUM
 #undef CP11X_BASE
 #undef CP11X_PCIEx_MEM_BASE
 #undef CP11X_PCIEx_MEM_SIZE
diff --git a/arch/arm64/boot/dts/marvell/cn9131-db.dts b/arch/arm64/boot/dts/marvell/cn9131-db.dts
index 3c975f98b2a3..408786cf8236 100644
--- a/arch/arm64/boot/dts/marvell/cn9131-db.dts
+++ b/arch/arm64/boot/dts/marvell/cn9131-db.dts
@@ -58,6 +58,7 @@
  */
 
 #define CP11X_NAME		cp1
+#define CP11X_NUM		1
 #define CP11X_BASE		f4000000
 #define CP11X_PCIEx_MEM_BASE(iface) (0xe2000000 + (iface * 0x1000000))
 #define CP11X_PCIEx_MEM_SIZE(iface) 0xf00000
@@ -68,6 +69,7 @@
 #include "armada-cp115.dtsi"
 
 #undef CP11X_NAME
+#undef CP11X_NUM
 #undef CP11X_BASE
 #undef CP11X_PCIEx_MEM_BASE
 #undef CP11X_PCIEx_MEM_SIZE
diff --git a/arch/arm64/boot/dts/marvell/cn9132-db.dts b/arch/arm64/boot/dts/marvell/cn9132-db.dts
index 4ef0df3097ca..c3124d849e3a 100644
--- a/arch/arm64/boot/dts/marvell/cn9132-db.dts
+++ b/arch/arm64/boot/dts/marvell/cn9132-db.dts
@@ -76,6 +76,7 @@
  */
 
 #define CP11X_NAME		cp2
+#define CP11X_NUM		2
 #define CP11X_BASE		f6000000
 #define CP11X_PCIEx_MEM_BASE(iface) (0xe5000000 + (iface * 0x1000000))
 #define CP11X_PCIEx_MEM_SIZE(iface) 0xf00000
@@ -86,6 +87,7 @@
 #include "armada-cp115.dtsi"
 
 #undef CP11X_NAME
+#undef CP11X_NUM
 #undef CP11X_BASE
 #undef CP11X_PCIEx_MEM_BASE
 #undef CP11X_PCIEx_MEM_SIZE
-- 
2.31.1

